src/Analysis/Codelyzer.Analysis.Common/MSBuildDetector.cs

```diff
@@ -80,24 +80,41 @@ private string GetVisualStudioVersionNumber(string solutionPath)
         public List<VisualStudioInstanceData> GetVisualStudioInstallations()
         {
             var visualStudioInstances = new List<VisualStudioInstanceData>();
-
-            var query = new SetupConfiguration();
-            var query2 = (ISetupConfiguration2)query;
-            var e = query2.EnumAllInstances();
-
-            int fetched;
-            var instances = new ISetupInstance[1];
-            do
+            try
             {
-                e.Next(1, instances, out fetched);
-                if (fetched > 0)
+                var query = new SetupConfiguration();
+                var query2 = (ISetupConfiguration2)query;
+                var e = query2.EnumAllInstances();
+
+                int fetched;
+                var instances = new ISetupInstance[1];
+                do
                 {
-                    var visualStudioInstance = ParseVisualStudioInstanceData(instances[0]);
-                    visualStudioInstances.Add(visualStudioInstance);
+                    e.Next(1, instances, out fetched);
+                    if (fetched > 0)
+                    {
+                        var visualStudioInstance = ParseVisualStudioInstanceData(instances[0]);
+                        visualStudioInstances.Add(visualStudioInstance);
+                    }
                 }
+                while (fetched > 0);
+            }
+            catch (Exception ex)
+            {
+                //Visual studio is not installed
+            }
+            if (!visualStudioInstances.Any())
```

Comment:
can this be null? if so we need to check for null as well befor checking for any

Reply:
This would not be null in this case because it is initialized as an empty list